### ISSUE TITLE
SCIP returns "Solver (asl) did not exit normally" when solving fo8_ar25_1 from MINLPLIB via Pyomo

### ISSUE THREAD

**Issue by LucasBoTang** [scipopt/scip#122]
## Issue Description

I encounter errors when using SCIP through Pyomo’s ASL interface to solve MINLPLIB instances, such as [`fo8_ar25_1`](https://www.minlplib.org/fo8_ar25_1.html). 

The solver terminates with `the Solver (asl) did not exit normally` error.

## Steps to Reproduce

I use the provided Pyomo model from MINLPLIB: [https://www.minlplib.org/py/fo8_ar25_1.py](https://www.minlplib.org/py/fo8_ar25_1.py).

And run the following Python code:
```python

# add fo8_ar25_1 model as the provided Pyomo script

from pyomo import opt as po
solver = po.SolverFactory("scip")
solver.solve(m, tee=True)
```

## Environment

- OS: Windows 11 Home (10.0.22631 N/A Build 22631)
- **Python:** 3.12.8
- **Pyomo:** 6.8.2
- **SCIP:** 9.2.0
- **AMPL/MP:** 690e9e7
- **Ipopt:** 3.14.16

## Error Message

The solver fails with the message:
```
SCIP version 9.2.0 [precision: 8 byte] [memory: block] [mode: optimized] [LP solver: Soplex 7.1.2] [GitHash: 74cea9222e]
Copyright (c) 2002-2024 Zuse Institute Berlin (ZIB)

External libraries:
  Soplex 7.1.2         Linear Programming Solver developed at Zuse Institute Berlin (soplex.zib.de) [GitHash: b040369c]
  CppAD 20180000.0     Algorithmic Differentiation of C++ algorithms developed by B. Bell (github.com/coin-or/CppAD)
  ZLIB 1.3.1           General purpose compression library by J. Gailly and M. Adler (zlib.net)
  GMP 6.3.0            GNU Multiple Precision Arithmetic Library developed by T. Granlund (gmplib.org)
  ZIMPL 3.6.2          Zuse Institute Mathematical Programming Language developed by T. Koch (zimpl.zib.de)
  AMPL/MP 690e9e7      AMPL .nl file reader library (github.com/ampl/mp)
  PaPILO 2.4.0         parallel presolve for integer and linear optimization (github.com/scipopt/papilo) (built with TBB) [GitHash: 2d9fe29f]
  Nauty 2.8.8          Computing Graph Automorphism Groups by Brendan D. McKay (users.cecs.anu.edu.au/~bdm/nauty)
  sassy 1.1            Symmetry preprocessor by Markus Anders (github.com/markusa4/sassy)
  Ipopt 3.14.16        Interior Point Optimizer developed by A. Waechter et.al. (github.com/coin-or/Ipopt)

user parameter file <scip.set> not found - using default parameters
read problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo.nl>
============

original problem has 180 variables (0 bin, 72 int, 0 impl, 108 cont) and 435 constraints

solve problem
=============

  [linear] <lc431>:  +0.5<x14>[C] (+1.8974) +0.5<x16>[C] (+1.8974) -<x104>[C] (+0) +<x106>[C] (+0) -12<i144>[I] (+0) -12<i179>[I] (+0) <= 0;
;
violation: right hand side is violated by 1.8974
all 1 solutions given by solution candidate storage are infeasible

presolving:
(round 1, fast)       57 del vars, 169 del conss, 0 add conss, 131 chg bounds, 1 chg sides, 2 chg coeffs, 0 upgd conss, 0 impls, 0 clqs
(round 2, fast)       57 del vars, 169 del conss, 0 add conss, 147 chg bounds, 1 chg sides, 2 chg coeffs, 0 upgd conss, 0 impls, 0 clqs
   (0.0s) probing cycle finished: starting next cycle
   (0.0s) symmetry computation started: requiring (bin +, int +, cont +), (fixed: bin -, int -, cont -)
   (0.0s) no symmetry present (symcode time: 0.00)
presolving (3 rounds: 3 fast, 1 medium, 1 exhaustive):
 57 deleted vars, 169 deleted constraints, 0 added constraints, 147 tightened bounds, 0 added holes, 1 changed sides, 0 changed coefficients
 8 implications, 0 cliques
presolved problem has 123 variables (1 bin, 70 int, 0 impl, 52 cont) and 266 constraints
    248 constraints of type <linear>
     18 constraints of type <nonlinear>
Presolving Time: 0.00

 time | node  | left  |LP iter|LP it/n|mem/heur|mdpt |vars |cons |rows |cuts |sepa|confs|strbr|  dualbound   | primalbound  |  gap   | compl.
  0.0s|     1 |     0 |   143 |     - |  2939k |   0 | 159 | 266 | 320 |   0 |  0 |   0 |   0 | 0.000000e+00 |      --      |    Inf | unknown
  0.0s|     1 |     0 |   521 |     - |  3285k |   0 | 159 | 266 | 382 |  62 |  1 |   0 |   0 | 0.000000e+00 |      --      |    Inf | unknown
  0.0s|     1 |     0 |   619 |     - |  3406k |   0 | 159 | 266 | 419 |  99 |  2 |   0 |   0 | 5.333268e-01 |      --      |    Inf | unknown
  0.0s|     1 |     0 |   713 |     - |  3531k |   0 | 159 | 266 | 447 | 127 |  3 |   0 |   0 | 3.339215e+00 |      --      |    Inf | unknown
  0.0s|     1 |     0 |   790 |     - |  3630k |   0 | 159 | 266 | 464 | 144 |  4 |   0 |   0 | 3.688542e+00 |      --      |    Inf | unknown
  0.0s|     1 |     0 |   851 |     - |  3986k |   0 | 159 | 266 | 487 | 167 |  5 |   0 |   0 | 6.671715e+00 |      --      |    Inf | unknown
  0.0s|     1 |     0 |   919 |     - |  4126k |   0 | 159 | 266 | 509 | 189 |  6 |   0 |   0 | 8.447069e+00 |      --      |    Inf | unknown
  0.0s|     1 |     0 |   982 |     - |  4318k |   0 | 159 | 266 | 523 | 203 |  7 |   0 |   0 | 9.261466e+00 |      --      |    Inf | unknown
  0.0s|     1 |     0 |  1065 |     - |  4407k |   0 | 159 | 266 | 549 | 229 |  8 |   0 |   0 | 1.209753e+01 |      --      |    Inf | unknown
  0.0s|     1 |     0 |  1113 |     - |  4535k |   0 | 159 | 266 | 572 | 252 |  9 |   0 |   0 | 1.275879e+01 |      --      |    Inf | unknown
  0.0s|     1 |     0 |  1180 |     - |  4727k |   0 | 159 | 266 | 592 | 272 | 10 |   0 |   0 | 1.369507e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1262 |     - |  4792k |   0 | 159 | 266 | 611 | 291 | 11 |   0 |   0 | 1.464301e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1290 |     - |  4858k |   0 | 159 | 266 | 626 | 306 | 12 |   0 |   0 | 1.488309e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1306 |     - |  4926k |   0 | 159 | 266 | 485 | 315 | 13 |   0 |   0 | 1.489459e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1319 |     - |  4972k |   0 | 159 | 266 | 488 | 318 | 14 |   0 |   0 | 1.489459e+01 |      --      |    Inf | unknown
 time | node  | left  |LP iter|LP it/n|mem/heur|mdpt |vars |cons |rows |cuts |sepa|confs|strbr|  dualbound   | primalbound  |  gap   | compl.
  1.0s|     1 |     0 |  1329 |     - |  5025k |   0 | 159 | 266 | 494 | 324 | 15 |   0 |   0 | 1.490109e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1338 |     - |  5078k |   0 | 159 | 266 | 498 | 328 | 16 |   0 |   0 | 1.490780e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1340 |     - |  5131k |   0 | 159 | 266 | 502 | 332 | 17 |   0 |   0 | 1.490780e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1437 |     - |  5212k |   0 | 159 | 266 | 502 | 332 | 18 |   0 |   5 | 1.495038e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1439 |     - |  5212k |   0 | 159 | 266 | 504 | 334 | 19 |   0 |   5 | 1.495038e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1439 |     - |  5218k |   0 | 159 | 266 | 504 | 334 | 20 |   0 |  10 | 1.545542e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1439 |     - |  5266k |   0 | 159 | 266 | 504 | 334 | 21 |   0 |  15 | 1.545542e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1441 |     - |  5266k |   0 | 159 | 266 | 504 | 334 | 22 |   0 |  20 | 1.753963e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1442 |     - |  5266k |   0 | 159 | 266 | 505 | 335 | 23 |   0 |  20 | 1.753963e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1443 |     - |  5272k |   0 | 159 | 266 | 505 | 335 | 24 |   0 |  25 | 1.753963e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1443 |     - |  5272k |   0 | 159 | 266 | 505 | 335 | 25 |   0 |  30 | 1.753963e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1443 |     - |  5272k |   0 | 159 | 266 | 505 | 335 | 26 |   0 |  35 | 1.753963e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1443 |     - |  5272k |   0 | 159 | 266 | 505 | 335 | 27 |   0 |  41 | 1.753963e+01 |      --      |    Inf | unknown
  1.0s|     1 |     0 |  1446 |     - |  5272k |   0 | 159 | 266 | 505 | 335 | 28 |   0 |  42 | 1.753963e+01 |      --      |    Inf | unknown
  1.0s|     1 |     2 |  1446 |     - |  5272k |   0 | 159 | 266 | 505 | 335 | 29 |   0 |  42 | 1.753963e+01 |      --      |    Inf | unknown
[nlpi.c:585] ERROR: Value -9.59179e-11 for parameter solver tolerance cannot be negative
[nlp.c:3203] ERROR: Error <-14> in function call
[nlp.c:4074] ERROR: Error <-14> in function call
[scip_nlp.c:558] ERROR: Error <-14> in function call
[heur_nlpdiving.c:2430] ERROR: Error <-14> in function call
[heur.c:1315] ERROR: Error <-14> in function call
[solve.c:368] ERROR: Error <-14> in function call
[solve.c:5300] ERROR: Error <-14> in function call
[scip_solve.c:2664] ERROR: Error <-14> in function call
[heur_alns.c:2639] ERROR: Error <-14> in function call
WARNING: Transformed variable <t_t_x25> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x17> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x16> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x15> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x14> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x13> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x12> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x11> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x10> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x9> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x8> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x7> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x6> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x5> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x4> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x3> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x2> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x1> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_x0> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Transformed variable <t_t_i130> not released when freeing SCIP problem <t_C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x25> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x17> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x16> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x15> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x14> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x13> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x12> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x11> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x10> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x9> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x8> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x7> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x6> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x5> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x4> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x3> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x2> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x1> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_x0> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
WARNING: Original variable <t_i130> not released when freeing SCIP problem <C:\Users\lucas\AppData\Local\Temp\tmpupj1l5qa.pyomo_alns_rens>.
[nlpi.c:585] ERROR: Value -0.0762652 for parameter solver tolerance cannot be negative
[nlp.c:3203] ERROR: Error <-14> in function call
[nlp.c:4074] ERROR: Error <-14> in function call
[scip_nlp.c:558] ERROR: Error <-14> in function call
[heur_nlpdiving.c:2430] ERROR: Error <-14> in function call
[heur.c:1315] ERROR: Error <-14> in function call
[solve.c:368] ERROR: Error <-14> in function call
[solve.c:5300] ERROR: Error <-14> in function call
[scip_solve.c:2664] ERROR: Error <-14> in function call
[scipshell.c:250] ERROR: Error <-14> in function call
[scipshell.c:305] ERROR: Error <-14> in function call
[scipshell.c:594] ERROR: Error <-14> in function call
SCIP Error (-14): the value is invalid for the given parameter
ERROR: Solver (scip) returned non-zero return code (4294967295)
ERROR: See the solver log above for diagnostic information.
Traceback (most recent call last):
  File "C:\OneDrive\Study\UofT\Research\DMIPL\minlplib\fo9_ar25_1.py", line 848, in <module>
    solver.solve(m, tee=True)
  File "C:\Users\lucas\anaconda3\envs\scip\Lib\site-packages\pyomo\opt\base\solvers.py", line 628, in solve
    raise ApplicationError("Solver (%s) did not exit normally" % self.name)
pyomo.common.errors.ApplicationError: Solver (scip) did not exit normally
```

## Additional Information

- No custom SCIP parameters are set (no `scip.set` file).
- The issue consistently occurs with fo8_ar25_1, and I have encountered similar errors with other MINLPLIB MINLP instances with **non-linear constraints**.

##Request

I would appreciate any guidance on whether this is a known issue, a bug in the SCIP-ASL interface, or a configuration problem. Suggestions on adjusting parameters or environment settings to avoid invalid parameter error would also be very helpful.

Thank you for your time and assistance.

**Comment by svigerske** [comment]
How was the SCIP executable be build? Which compilers and compiler versions?

**Comment by LucasBoTang** [comment]
The SCIP executable was installed via Conda from the `conda-forge` channel:  
[https://anaconda.org/conda-forge/scip](https://anaconda.org/conda-forge/scip)  

Since this is a precompiled binary provided by Conda-forge, the build was done by their maintainers. Unfortunately, I don't have direct control over the compilers and compiler versions used for the build.

**Comment by svigerske** [comment]
Ah ok, then that seems to be MSVC 19.29.30154.0 (MSVS 2022) (https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1081304&view=logs&jobId=fe7faf59-69bb-5487-b8fd-a35a7bda3a51&j=fe7faf59-69bb-5487-b8fd-a35a7bda3a51&t=55731638-dd49-5144-2129-5101b9dd1960). We'll need to see whether we can reprodcue and debug this.

You may want to check whether SCIP from https://github.com/scipopt/scip/releases/download/v920/SCIPOptSuite-9.2.0-win64.exe works better.

**Comment by svigerske** [comment]
I have no luck here.

The problem is not with ASL (which SCIP doesn't use) and very unlikely due to SCIP's reader for .nl files.
The log looks rather as if the initialization of `SCIP_NLPPARAM` member `solvertol` at
https://github.com/scipopt/scip/blob/d859d5fb839c9b90394c9e0afea296b7a32df507/src/scip/type_nlpi.h#L106
(when using [macro `SCIPsolveNLP()`](https://github.com/scipopt/scip/blob/d859d5fb839c9b90394c9e0afea296b7a32df507/src/scip/scip_nlp.h#L362-L363) in `heur_nlpdiving.c`) did not work for some reason.

We have own builds with MSVS 2019 and cl 19.29.30154.0, but I couldn't reproduce the issue with https://www.minlplib.org/osil/fo9_ar25_1.osil as input.
Also trying the code in question in isolation ([122.c.txt](https://github.com/user-attachments/files/18062905/122.c.txt)
[122a.c.txt](https://github.com/user-attachments/files/18062907/122a.c.txt)) didn't help.

I haven't tried with the executable from conda. But even if I could reproduce the issue with it, I wouldn't know how to debug it.

The log of the conda build,
```
-- The C compiler identification is MSVC 19.29.30154.0
-- The CXX compiler identification is MSVC 19.29.30154.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.29.30133/bin/HostX64/x64/cl.exe - skipped
```
looks a bit suspicious to me.
19.29.30154.0 is MSVS 2019, but then there are frequent mentions of MSVS 2022. I won't say that this mix is likely responsible for this issue, but wouldn't exclude it either. I don't think that the [conda-forge/scipoptsuite-feedstock](https://github.com/conda-forge/scipoptsuite-feedstock/) project has control over the compiler choice and setup. CC @AntoinePrv just in case.

**Comment by LucasBoTang** [comment]
Thank you so much for your detailed response and assistance!

I wanted to let you know that the issue has now been resolved. I noticed that my `SCIP` path pointed to  `AMPL/solvers`. And I removed both `AMPL` and `SCIP` from my system and deleted the PATH in environment variables. Then, I reinstalled SCIP using the .exe installer and updated PATH. After doing this, the problem was resolved.

Thank you again for your time and insights! It really helped me narrow down the potential causes!
